src/applications/disability-benefits/2346/config/form.js

```diff
@@ -69,27 +67,75 @@ const submit = (form, formConfig) => {
     useTemporaryAddress,
   };
 
-  const eventData = {
-    'bam-quantityOrdered': itemQuantities,
+  const options = {
+    body: JSON.stringify(payload),
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
   };
 
-  const body = JSON.stringify(payload);
-  return submitToUrl(
-    body,
-    formConfig.submitUrl,
-    trackingPrefix,
-    eventData,
-  ).catch(() => {
-    recordEvent({
-      event: `${trackingPrefix}-submission-failed`,
-      ...eventData,
+  const onSuccess = resp => {
+    const successfulSubmissions = resp.filter(response =>
+      response.status.toLowerCase().includes('processed'),
+    );
+    const failedSubmissions = resp.filter(
+      response => !response.status.toLowerCase().includes('processed'),
+    );
+    const successfulSubmissionProductIds = successfulSubmissions.map(
+      submission => submission.productId,
+    );
+    const failedSubmissionProductIds = failedSubmissions.map(
+      submission => submission.productId,
+    );
+
+    // Case where all of the products were successfully ordered
+    if (successfulSubmissions.length > 0 && failedSubmissions.length === 0) {
+      recordEvent({
+        event: 'bam-2346a--submission-successful',
+        'bam-quantityOrdered': itemQuantities,
+      });
+    }
+
+    // For partially successful orders we want to send all of the productIds
+    if (failedSubmissions.length && successfulSubmissions.length) {
+      recordEvent({
+        event: 'bam-2346a--submission-successful',
+        'partial-failed': true,
+        'product-ids-successful': successfulSubmissionProductIds.join(' '),
+        'product-ids-failed': failedSubmissionProductIds.join(' '),
+      });
+    }
+    // Failed submissions still return a 200 response so we need to ensure we
+    // still submit a submission failed event if none of the items ordered were successful
```

Comment:
This logic of a failed response sending a 200 is by design so we can inject our own error messaging on the confirmation page